src/System.Net.Http.WinHttpHandler/tests/UnitTests/TestServer.cs

```diff
@@ -169,5 +154,29 @@ public static void Reset()
             responseHeaders = null;
             dataAvailablePercentage = 1.0;
         }
+
+        public static byte[] CompressBytes(byte[] bytes, bool useGZip)
+        {
+            byte[] compressedBytes;
+
+            using (var memoryStream = new MemoryStream())
+            {
+                Stream compressedStream = null;
+                if (useGZip)
+                {
+                    compressedStream = new GZipStream(memoryStream, CompressionMode.Compress);
+                }
+                else
+                {
+                    compressedStream = new DeflateStream(memoryStream, CompressionMode.Compress);
+                }
+
+                compressedStream.Write(bytes, 0, bytes.Length);
+                compressedStream.Dispose();
+                compressedBytes = memoryStream.ToArray();
```

Comment:
Is there any reason not to `return memoryStream.ToArray()` and avoid creating compressedBytes?

Reply:
Good catch. I was refactoring that code a lot before I submitted the PR. I missed cleaning that up.